meteor/server/api/showStyles.ts

```diff
@@ -75,50 +75,94 @@ export async function insertShowStyleBaseInner(organizationId: OrganizationId |
 		_rundownVersionHash: '',
 	}
 	ShowStyleBases.insert(showStyleBase)
-	await insertShowStyleVariantInner(showStyleBase, 'Default')
+
+	const showStyleVariant: ShowStyleVariant = {
+		_id: getRandomId(),
+		name: 'Default',
+		blueprintConfigWithOverrides: wrapDefaultObject({}),
+		_rundownVersionHash: '',
+		_rank: 500,
+		showStyleBaseId: showStyleBase._id,
+	}
+	await insertShowStyleVariant(showStyleVariant)
 	return showStyleBase._id
 }
-export async function insertShowStyleVariant(
-	context: MethodContext | Credentials,
-	showStyleBaseId: ShowStyleBaseId,
-	name?: string
-): Promise<ShowStyleVariantId> {
+async function assertShowStyleBaseAccess(context: MethodContext | Credentials, showStyleBaseId: ShowStyleBaseId) {
 	check(showStyleBaseId, String)
 
 	const access = await ShowStyleContentWriteAccess.anyContent(context, showStyleBaseId)
 	const showStyleBase = access.showStyleBase
 	if (!showStyleBase) throw new Meteor.Error(404, `showStyleBase "${showStyleBaseId}" not found`)
-
-	return insertShowStyleVariantInner(showStyleBase, name)
 }
-export async function insertShowStyleVariantInner(
-	showStyleBase: ShowStyleBaseLight,
+
+export async function createDefaultShowStyleVariant(
+	context: MethodContext | Credentials,
+	showStyleBaseId: ShowStyleBaseId,
 	name?: string
 ): Promise<ShowStyleVariantId> {
-	return ShowStyleVariants.insertAsync({
+	await assertShowStyleBaseAccess(context, showStyleBaseId)
+
+	const highestRank = ShowStyleVariants.find(
+		{},
```

Comment:
As a general rule; never do full-db-searches, this is a classic source for performance problems!
This should be limited to `{ showStyleBaseId }`.

Reply:
Adding to that, it is loading the full document of every showstyle variant, when it could be doing a findOne, and only fetching the `_rank` field of the matching document:
```
ShowStyleVariants.findOneAsnc({
  showStyleBaseId
}, {
  fields: {
    _rank: 1
  },
  sort: {
    _rank: -1,
  },
})
```